packages/rest/src/sequence.ts

```diff
@@ -116,3 +116,77 @@ export class DefaultSequence implements SequenceHandler {
     }
   }
 }
+
+/**
+ * A sequence implementation using middleware chains
+ */
+export class MiddlewareSequence implements SequenceHandler {
+  /**
+   * Constructor: Injects findRoute, invokeMethod & logError
+   * methods as promises.
+   *
+   * @param invokeMiddleware - invoker for registered middleware in a chain.
+   * To be injected via SequenceActions.INVOKE_MIDDLEWARE.
+   */
+  constructor(
+    @inject(SequenceActions.INVOKE_MIDDLEWARE)
+    readonly invokeMiddleware: InvokeMiddleware,
+    @config()
+    readonly options: InvokeMiddlewareOptions[] = [],
+  ) {
+    if (this.options.length === 0) {
+      this.options = [
+        {
+          chain: 'middlewareChain.rest',
+          orderedGroups: [
+            // Please note that middleware is cascading. The `SendResponse` is
+            // added first to invoke downstream middleware to get the result or
+            // catch errors so that it can produce the http response.
+            'sendResponse',
+
+            // default
+            'cors',
+            'apiSpec',
+            'middleware',
+
+            'authentication',
+
+            // rest
+            'findRoute',
+            'parseParams',
+            'invokeMethod',
+          ],
+        },
+      ];
+    }
+  }
+
+  /**
+   * Runs the default sequence. Given a handler context (request and response),
+   * running the sequence will produce a response or an error.
+   *
+   * Default sequence executes these steps
+   *  - Finds the appropriate controller method, swagger spec
+   *    and args for invocation
+   *  - Parses HTTP request to get API argument list
+   *  - Invokes the API which is defined in the Application Controller
+   *  - Writes the result from API into the HTTP response
+   *  - Error is caught and logged using 'logError' if any of the above steps
```

Comment:
`logError` is not included in the options

Reply:
also how is an error redirected the `LogErrorProvider`, is that logic part of the new express package ?

Reply:
logError/LogErrorProvider are not middleware. They are injected into `SendResponseMiddleware` to log errors.